### DISCUSSION TITLE
Custom Checkout

### DISCUSSION THREAD

**Discussion by Raigeku** [swellstores/community#577] [Help] [answered]
Hi, I'm kind of unexperienced in this ecommerce field, so I'm asking for a little guide here, having tried to comprehend it from the developer wiki but failed.
I built this custom checkout page with a form that gets billing info, what payment method and what delivery to use, notes etc etc.
There is a button to submit the form, now exactly what should be the flow of things using Swell.js with Next?
If I choose to pay with Stripe for example what should be the course of action using the Swell API?
Thanks

**Comment by ericingram** [comment]
Check out these docs: https://developers.swell.is/frontend-api/payments#stripe-klarna-ideal-bancontact-integration

**Comment by Raigeku** [comment]
Maybe I've been inaccurate, I'll try to explain myself better. Let's say I have a cart with some products on it, what should be the process? Update the cart with the billing info and then submit the order? I rendered a stripe card element on the summary form page, where the client gives all his infos to process the order. For example, how do I process the payment, will it process itself as the order is submitted or do I need to do anything else?

**Reply by ericingram** [reply, marked as answer]
@Raigeku that's correct, you update the cart with the billing info and then create an order with the cart ID using either the frontend or backend API.

Submitting an order from a cart is done with an HTTP request like this:

```
POST /orders { "cart_id": "<id>" }
```

Or using the frontend API, like this: https://developers.swell.is/frontend-api/orders#submit-an-order

Once the cart has the correct billing info, it will charge the card immediately or not depending on your settings

**Reply by Raigeku** [reply]
Ok, seems good enough, one last question. For the payment authorization ( OTP ) that every bank has, how does it work? Will the bank page open by itself once you try to submit the order to authorize it?

**Reply by ericingram** [reply]
I think it depends on the bank integration through Stripe, so the application should be able respond to it in any case.

@vstudenichnik-insoft do you know how this works?

**Reply by vstudenichnik-insoft** [reply]
@Raigeku It depends on the payment method. Swell supports the following payment methods via Stripe:
- Card, Google Pay, Apple Pay – payment authorization occurs in a payment confirmation modal.
- iDEAL, Bancontact, Klarna – redirects the user to the bank’s page.

[Here's an example](https://developers.swell.is/frontend-api/payments#tokenize-elements) of integrating card payments via Stripe

**Reply by Raigeku** [reply]
Good enough, thank you guys!
